lstests/core/service.py

```diff
@@ -0,0 +1,1558 @@
+        self.print_config()
+        self.launch_cluster()
+        self.wait_nodes_ready()
+        self.schedule_protocol_feature_activations()
+        self.set_bios_contract()
+        if not dont_newacco:
+            self.bios_create_accounts_in_parallel(accounts=self.producers)
+            if not dont_setprod:
+                self.set_producers()
+        self.check_sync()
+        for t in self.verify_threads:
+            t.join()
+        self.info(">>> [Launch] ----------------------- END -----------------------------------------------------------")
+
+# --------------- start-up and shut-down ------------------------------------------------------------------------------
+
+    def launch_cluster(self, **call_kwargs):
+        return self.call("launch_cluster",
+                         node_count=self.node_count,
+                         nodes=self.nodes,
+                         shape=self.topology,
+                         center_node_id=self.center_node_id,
+                         extra_configs=self.extra_configs,
+                         extra_args=self.extra_args,
+                         **call_kwargs)
+
+    def stop_cluster(self, **call_kwargs):
+        return self.call("stop_cluster", **call_kwargs)
+
+    def clean_cluster(self, **call_kwargs):
+        return self.call("clean_cluster", **call_kwargs)
+
+    def start_node(self, node_id, extra_args=None, **call_kwargs):
+        return self.call("start_node", node_id=node_id, extra_args=extra_args, **call_kwargs)
+
+    def stop_node(self, node_id, kill_sig=15, check=True, dont_raise=False, **call_kwargs):
+        """kill_sig: 15 for soft kill, 9 for hard kill"""
+        cx = self.call("stop_node", node_id=node_id, kill_sig=kill_sig, **call_kwargs)
+        if check:
+            for _ in range(10):
+                if self.is_node_down(node_id=node_id):
+                    return cx
+                time.sleep(1)
+            else:
+                if not dont_raise:
+                    raise BlockchainError(f"Node {node_id} cannot be properly stopped by signal {kill_sig}.")
+        else:
+            return cx
+
+# --------------- simple queries: queries that are made directly via call() -------------------------------------------
+
+    def get_cluster_running_state(self, **call_kwargs):
+        return self.call("get_cluster_running_state", **call_kwargs)
+
+    def get_cluster_info(self, **call_kwargs):
+        return self.call("get_cluster_info", **call_kwargs)
+
+    def get_info(self, node_id, **call_kwargs):
+        return self.call("get_info", node_id=node_id, **call_kwargs)
+
+    def get_block(self, block_num_or_id, node_id=0, **call_kwargs):
+        return self.call("get_block", node_id=node_id, block_num_or_id=block_num_or_id, **call_kwargs)
+
+    def get_account(self, name, node_id=0, **call_kwargs):
+        return call("get_account", node_id=node_id, name=name, **call_kwargs)
```

Comment:
need to add "self." here and in get_protocol_features

Reply:
should be fixed. can you check it again?

Reply:
Fixed now. So is that method not called yet, since it didn't have an execution error?